platform/legacy/LegacyPlatformProxifier.ts

```diff
@@ -0,0 +1,87 @@
+import { Server } from 'net';
+import { IncomingMessage, OutgoingMessage } from 'http';
+import { EventEmitter } from 'events';
+
+import { Logger } from '../logging';
+
+export class LegacyPlatformProxifier extends EventEmitter {
+  private server: Server;
+  private readonly eventHandlers: any;
+
+  constructor(
+    private readonly log: Logger,
+    readonly kbnServer: any,
+    private readonly startPlatform: () => Promise<any>,
+    private readonly stopPlatform: () => Promise<any>
+  ) {
+    super();
+
+    const forwardEvent = (name: string, ...args: any[]) => {
+      this.log.info(`Event is being forwarded: ${name}`);
+      this.emit(name, ...args);
+    };
+
+    // HapiJS expects that the following events will be generated by `listener`, see:
+    // https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js.
+    this.eventHandlers = {
+      listening: forwardEvent.bind(this, 'listening'),
+      error: forwardEvent.bind(this, 'error'),
+      clientError: forwardEvent.bind(this, 'clientError'),
+      connection: forwardEvent.bind(this, 'connection')
+    };
+  }
+
+  address() {
+    return this.server && this.server.address();
+  }
+
+  async listen(port: number, host: string, callback: Function) {
+    this.log.info(`"listen" has been called (${host}:${port}).`);
+
+    await this.startPlatform();
+
+    callback && callback();
+  }
+
+  async close(callback: Function) {
```

Comment:
Maybe:

```js
async close(callback: () => {}) {
```

Reply:
you mean `() => void` right? If so, then ✔️